proposals/0345-android-12-issues.md

```diff
@@ -165,19 +268,67 @@ private static SdlDeviceListener getSdlDeviceListener(Context context, Bluetooth
+If a phone connects to the head unit over USB we can still start the `SdlRouterService`, but if the designated app does not have Bluetooth permission, the `bluetoothTransport` will still not be able to start. This could cause confusion for the user if they expect the `SdlRouterService` to connect over USB and Bluetooth but the `SdlRouterService` will only connect over USB. The suggested solution is to present a notification to the user reminding them to enable Bluetooth permission. By clicking on the notification the user will be directed to the app's settings page to grant this permission. Meanwhile the `SdlRouterService` will wait to initialize the `bluetoothTransport` and will continuously check the permission status. Once the permissions are granted, the `bluetoothTransport` will be started.
+
+Additionally the `MultiplexBluetoothTransport.java` class is using `BluetoothAdapter.cancelDiscovery()`, This requires an additional Runtime Permission of `BLUETOOTH_SCAN`. There is only one use case where this will be called and that is when the the `SdlRouterService` receives a `ROUTER_REQUEST_BT_CLIENT_CONNECT` message. This is never used in a production setting and as such we can check for the `BLUETOOTH_SCAN` permission when it is called but we do not need to require developers to have this permission.
 
 ##### SdlRouterService.java
 ~~~ java
+
+//...
+
+private final static int BT_PERMISSIONS_CHECK_FREQUENCY = 1000;
+
+
+//...
+
+// Check for BLUETOOTH_SCAN in the RouterHandler
+static class RouterHandler extends Handler {
+
+    //...
+
+    @Override
+    public void handleMessage(Message msg) {
+
+    //...
+
+    switch (msg.what) {
+        case TransportConstants.ROUTER_REQUEST_BT_CLIENT_CONNECT:
+            if (!AndroidTools.isBtScanPermissionGranted(service.getApplicationContext(), service.getPackageName())) {
+                break;
+            }
+     //...
+
+//...
+
 private boolean initCheck() {
 
     //...
 
     // If Android 12 or newer and we do not have Bluetooth runtime permissions
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S && !AndroidTools.areBtPermissionsGranted(this, this.getPackageName())) {
-        //If the intent to start the RouterService came from the USB Activity
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S && !AndroidTools.isBtConnectPermissionGranted(this, this.getPackageName())) {
+        if (!isConnectedOverUSB) {
+            //If the intent to start the RouterService came from somewhere other than the USB Activity
+            //i.e. an SDL APP on SDL Android 4.11 or older tries to start this RouterService while the device is connected over Bluetooth
+            return false;
+        }
+    }
+
+    //...
+
+    return true;
+
+}
+
+//...
+
+public void onTransportConnected(final TransportRecord record) {
+
+//...
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S && !AndroidTools.isBtConnectPermissionGranted(this, this.getPackageName())) {
```

Comment:
Can you please advise for which requested revision these changes were made? Is this part of 10, alongside line 340?